### PR TITLE
Only update position if document is showing

### LINKED ISSUE #48: Restrict access to visible and user-activation-received frames
Geolocation functionality poses clear benefits and risks to user privacy.  In order to better distinguish the former from the latter, and prevent location from becoming an unknown ongoing tracking vector, the spec should require signals that the user is intentionally interacting with a frame before the frame can request location data (even for a frame that was previously given location permission).

Two common signals that should be incorporated into the spec are that frames need to be visible, and to have received a user activation before they can access location information, even if the frame was previously given permission.

### PR DESCRIPTION
**Pull request by marcoscaceres**
Closes #48

The following tasks have been completed:

 * [ ] Modified Web platform tests (link to pull request)

Implementation commitment:

 * [X] WebKit (https://bugs.webkit.org/show_bug.cgi?id=)
 * [X] Chromium (https://bugs.chromium.org/p/chromium/issues/detail?id=)
 * [ ] [Gecko](https://bugzilla.mozilla.org/show_bug.cgi?id=1653549)

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/geolocation-api/pull/67.html" title="Last updated on Mar 24, 2021, 2:53 AM UTC (5c7c7e0)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/geolocation-api/67/6256b91...5c7c7e0.html" title="Last updated on Mar 24, 2021, 2:53 AM UTC (5c7c7e0)">Diff</a>

### CODE REVIEW

**Comment by marcoscaceres** [comment]
Sent a patch to Gecko as part of https://bugzilla.mozilla.org/show_bug.cgi?id=1653549

File: index.html [hunk 1]
```diff
@@ -546,11 +546,16 @@ <h3>
...
-                    <li>If the <dfn>new position differs significantly from the
-                    previous position</dfn>, invoke the |successCallback| with
-                    a new {{GeolocationPosition}} object that reflects the
-                    result of the acquisition operation. This step MAY be
-                    subject to callback [=rate limitation=].
+                    <li>Let |document:Document| be <a>current settings
+                    object</a>'s [=environment settings object / responsible
+                    document=].
+                    </li>
+                    <li>If |document| is [=Document/visible=] and the <dfn>new
```
Comment (marcoscaceres) [inline index.html:553]: @reillyeon ... Alternative was just:

> If |document| is [=Document/hidden=], return. 

Though I feel this was more clear.

Comment (marcoscaceres) [reply index.html:553]: Actually, I'm going to flip it...

**Comment by marcoscaceres** [comment]
I just realized that this is not enough... I'll update this and send again.

**Comment by marcoscaceres** [comment]
Ok, did a deep dive and did some more testing. This requires a fairly hefty rewrite of the spec.

Basically, we need to wire up visibility change listeners for when the methods are called. All the timing stuff is also affected, because the position request happens after the document becomes visible again.

**Comment by marcoscaceres** [comment]
will send again after rewrite...

**Comment by reillyeon** [comment]
> All the timing stuff is also affected, because the position request happens after the document becomes visible again.

Just for the record, Chromium doesn't implement the timing requirements around permissions but I think that should qualify as a bug as the requirements make sense.

**Comment by marcoscaceres** [comment]
As part of exploring of adding this to Gecko,  Edgar [identified some inconsistency](https://bugzilla.mozilla.org/show_bug.cgi?id=1653549#c8) around how visibility is handled between Chrome and Safari... I'll report back once I have a better handle on what's going on.

**Pull request closed without merging**